projboard/models/user.py

```diff
@@ -22,11 +21,11 @@ def get_user_by_nickname(nickname):
         :param nickname: the nickname to search
         :return: User object
         """
-        try:
-            user = User.objects.get(nickname=nickname)
-        except ObjectDoesNotExist:
-            return None
-        return user
+        user = User.objects.get(nickname=nickname)
+        if user not in User.objects.all():
+            raise User.DoesNotExist
+        else:
+            return user
```

Comment:
I think that `User.objects.get(nickname=nickname)` will raise an Exception if not found user with the nickname that we searching.

Maybe you can write it like that:

```
user = User.objects.get(nickname=nickname)

return user

```
The method will raise `User.DoesNotExist`  if not found user with this nickname.

You can read about the DoesNotExist Exception [here](https://docs.djangoproject.com/en/4.1/topics/db/queries/#retrieving-a-single-object-with-get)

Reply:
> I think that `User.objects.get(nickname=nickname)` will raise an Exception if not found user with the nickname that we searching.
> 
> Maybe you can write it like that:
> 
> ```
> user = User.objects.get(nickname=nickname)
> 
> return user
> ```
> 
> The method will raise `User.DoesNotExist` if not found user with this nickname.
> 
> You can read about the DoesNotExist Exception [here](https://docs.djangoproject.com/en/4.1/topics/db/queries/#retrieving-a-single-object-with-get)

True it returns its own exception, fixed